packages/create-styles/src/cssCustomProperties/createRootStore.js

```diff
@@ -0,0 +1,97 @@
+import { getPropValue } from './getPropValue';
+import { hasVariable, isCustomProperty } from './utils';
+
+/**
+ * Stores CSS config variables that are expected to be added to :root.
+ * This is used for CSS variable fallback handling for IE 11.
+ */
+class RootStore {
+	state = {};
+
+	constructor(initialState = {}) {
+		this.setState(initialState);
+	}
+
+	/**
+	 * Retrieve a value from the state.
+	 * @param {string} key The key to retrieve.
+	 * @returns {string} The value.
+	 */
+	get = (key) => this.state[key];
+
+	/**
+	 * Retrieves the current state.
+	 * @returns {object} The state.
+	 */
+	getState = () => ({ ...this.state });
+
+	/**
+	 * Sets the state.
+	 * @param {object} next The next state to merge into the current state.
+	 * @returns {object} The state.
+	 */
+	setState = (next = {}) => {
+		this._updateState(next);
+		this._resolveVariablesInStateValue();
+
+		return this.state;
+	};
+
+	/**
+	 * Updates the state.
+	 * @param {object} next The next state to merge into the current state.
+	 */
+	_updateState = (next = {}) => {
+		this.state = { ...this.state, ...next };
+	};
+
+	/**
+	 * Resolves potential CSS variables that may exist within the state's values.
+	 * This function cycles a maximum of 3 times to (hopefully) resolve chaining variable references.
```

Comment:
Why does this limitation exist? Why can't we just follow the chain until we reach a concrete value rather than a variable?

Reply:
@saramarcondes I gave this some thought! I think you're right 👍 .

I don't think there's a scenario where there may be stray / unresolvable CSS variable values.
The reason is because this resolver runs against the base G2 config values, which should alway compute to something.

It (currently) does not attempt to resolve custom theme values (which may contain unresolvable values).

Since this initial pass is to support base-level rendering for IE11, I think this is okay.

Thank you!!!